packages/graphql/src/translate/utils/resolveTree.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ResolveTree } from "graphql-parse-resolve-info";
+import { BaseField } from "../../types";
+import { removeDuplicates } from "../../utils/utils";
+
+/* Finds a field of selection based on field name */
+export const fieldNameExistsInSelection = (selection: Record<string, ResolveTree>) => (fieldName: string) =>
+    Object.values(selection).find((resolveTree) => resolveTree.name === fieldName);
```

Comment:
**question:** Are we getting all the value from curry functions here?

I feel we could be achieving the same with a much simpler function:

```suggestion
export function fieldNameExistsInSelection(selection: Record<string, ResolveTree>, fieldName: string): ResolveTree {
    return Object.values(selection).find((resolveTree) => resolveTree.name === fieldName);
}
```

Reply:
In one iteration of the refactor I was reusing this function, but at the end it makes more sense to have an uncurried function here.

Reply:
Refactored in https://github.com/neo4j/graphql/pull/842/commits/59e8d9b2848938364adfaf1402f3494d3071182a